### PR TITLE
fix (many) shellcheck warnings

### PR DESCRIPTION
**Pull request by jakobhellermann**
Related issue: #1107 

The first commit fixes the 'machine applicable' shellcheck warnings, the second one fixes some manually.

There are still some warnings left:

```
In build_sysroot/prepare_sysroot_src.sh line 24:
for file in $(ls ../../patches/ | grep -v patcha); do
              ^-- SC2010: Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.
```
I wasn't sure why you were ignoring 'patcha' here. Is this still needed?

```
In scripts/config.sh line 31:
      RUN_WRAPPER='wine'
      ^---------^ SC2034: RUN_WRAPPER appears unused. Verify use (or export if used externally).
```
Was this supposed to be used somewhere?

```
In scripts/rustup.sh line 30:
	cg_clif=$(pwd)
        ^-----^ SC2034: cg_clif appears unused. Verify use (or export if used externally).
```
Same as above.

```
In scripts/tests.sh line 102:
    cat examples/regexdna-input.txt | ../build/cargo.sh run --example shootout-regex-dna | grep -v "Spawned thread" > res.txt
        ^-------------------------^ SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
```
I think not using cat here makes readability worse.

### CODE REVIEW

File: scripts/cargo.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ cmd=$1
 shift || true
 
 if [[ "$cmd" = "jit" ]]; then
-cargo "+${TOOLCHAIN}" rustc $@ -- --jit
+cargo "+${TOOLCHAIN}" rustc "$@" -- --jit
```
Comment (bjorn3) [inline scripts/cargo.sh:13]: The old behavior is deliberate. Separate arguments to `cargo.sh` should be passed as separate arguments to cargo.

File: scripts/config.sh [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#!/bin/bash
```
Comment (bjorn3) [inline scripts/config.sh:1]: This file should never be executed, only sourced. Please remove this line and revert permissions back.

File: test.sh [hunk 3]
```diff
@@ -3,13 +3,13 @@ set -e
 
 export RUSTFLAGS="-Zrun_dsymutil=no"
 
-./build.sh --without-sysroot $@
+./build.sh --without-sysroot "$@"
```
Comment (bjorn3) [inline test.sh:6]: Same re separate arguments.

File: scripts/config.sh [hunk 4]
```diff
@@ -51,7 +53,8 @@ if [[ $(uname) == 'Darwin' ]]; then
    export RUSTFLAGS="$RUSTFLAGS -Clink-arg=-undefined -Clink-arg=dynamic_lookup"
 fi
 
-export LD_LIBRARY_PATH="$dir:$(rustc --print sysroot)/lib:$dir/target/out:$dir/sysroot/lib/rustlib/$TARGET_TRIPLE/lib"
+LD_LIBRARY_PATH="$dir:$(rustc --print sysroot)/lib:$dir/target/out:$dir/sysroot/lib/rustlib/$TARGET_TRIPLE/lib"
+export LD_LIBRARY_PATH
```
Comment (bjorn3) [inline scripts/config.sh:57]: I slightly prefer the old version.

**Review by bjorn3 — CHANGES_REQUESTED** [review]
Thanks!

> I wasn't sure why you were ignoring 'patcha' here. Is this still needed?

It allowed for quick disabling of specific patches by renaming them. I haven't used it in months, but it could be useful to keep.

> Was this supposed to be used somewhere?

`config.sh` is sourced by several files. `RUN_WRAPPER` is used by `tests.sh`.

> I think not using cat here makes readability worse.

I agree

File: scripts/cargo.sh (hunk 1 above)
Comment (jakobhellermann) [reply scripts/cargo.sh:13]: Actually I think this is correct: https://stackoverflow.com/a/3990540

"$@" expands to "$1" "$2" "$3".

The difference between `$@` and `"$@"` occurs when doing something like `./script "*" b c`, where the quoted variant will pass the * and the unquoted one will expand it.

File: scripts/config.sh (hunk 2 above)
Comment (jakobhellermann) [reply scripts/config.sh:1]: same as above

File: scripts/config.sh (hunk 4 above)
Comment (jakobhellermann) [reply scripts/config.sh:57]: I think the motivation was that if `rustc --print sysroot` fails, ` set -e` will not catch it inside an export, but I can change it.

File: scripts/config.sh (hunk 2 above)
Comment (jakobhellermann) [reply scripts/config.sh]: Done :+1:

File: scripts/cargo.sh (hunk 1 above)
Comment (bjorn3) [reply scripts/cargo.sh:13]: Didn't know that.

**Review by bjorn3 — APPROVED** [review]
Thanks!

**Pull request merged**